contracts/ZecaleDispatcher.sol

```diff
@@ -118,56 +114,53 @@ contract ZecaleDispatcher
         // Create an array to reuse to pass the nested inputs to the
         // application (note that the result is not passed, hence -1).
         uint256[] memory nested_proof_inputs =
-            new uint256[](_inputs_per_nested_tx - 1);
+            new uint256[](_inputs_per_nested_tx);
+
+        // Result bits
+        uint256 results = inputs[3];
 
         // Pass the details of each valid proof to the application
         for (uint256 nested_tx_idx = 0; nested_tx_idx < batch_size;
              ++nested_tx_idx) {
 
+            uint256 result = results & 0x1;
+            results = results >> 1;
+
+            // Skip nested transactions whose proofs are invalid.
+            // emit log("result", result);
+            if (result == 0) {
+                continue;
+            }
+
             // Note that the offsets here are all based on the assumption that
             // each scalar consists of 2 uint256 words, and that each nested
             // input is held in the final uint256 word.
 
-            // Of the inputs for this nested tx, the first `inputs_per_batch -
-            // 1` are the inputs to the nested proof. The final entry is the
-            // `result` for the nested proof.
-
             // Word index (in `inputs`) of the start of the inputs for this
-            // nested tx, offset by scalar_size_in_words - 1 to extract the LO
-            // word.
+            // nested tx. +1 to target LO word.
             uint256 batch_start_word_idx =
                 scalar_size_in_words *
                     (2 + inputs_per_nested_tx * nested_tx_idx)
-                - 1;
-            uint256 result_word_idx =
-                batch_start_word_idx +
-                (scalar_size_in_words * (inputs_per_nested_tx - 1));
-
+                + 1;
```

Comment:
Hum not sure I get the diffs here. I guess `2` is necessary to shift the vk scalars (does that need to be in the parenthesis to be further multiplied by `scalar_size_in_words`?), and the `+1` comes from the fact the the `result` is assumed to be stored "before" the nested tx inputs, right?

Reply:
Not quite.
```
    scalar_size_in_words *                        
        (2 + inputs_per_nested_tx * nested_tx_idx)
```
computes the offset (in words) of the start of the inputs for this nested tx.  +1 then selects the LO word, which is what we care about.  Hopefully that's clear, with the updated comment showing the layout (& the comment immediately above explaining +1).

Reply:
I see, makes sense. Thanks for updating the comments